parser.ts

```diff
@@ -759,44 +745,46 @@ export class Parser {
     return new Until(condition);
   }
 
-  private cd(): Instr {
-    const params = this.params();
-    return new Command('cd', params);
-  }
-
-  private cp(): Instr {
-    const params = this.params();
-    return new Command('cp', params);
-  }
-
-  private rm(): Instr {
-    const params = this.params();
-    return new Command('rm', params);
-  }
-
-  private touch(): Instr {
-    const params = this.params();
-    return new Command('touch', params);
-  }
-
-  private mv(): Instr {
-    const params = this.params();
-    return new Command('mv', params);
-  }
-
-  private mkdir(): Instr {
-    const params = this.params();
-    return new Command('mkdir', params);
+  private checkPrimaryStart(): boolean {
+    return (
+      this.check(TokenKind.DecimalLiteral) ||
+      this.check(TokenKind.HexLiteral) ||
+      this.check(TokenKind.OctalLiteral) ||
+      this.check(TokenKind.BinaryLiteral) ||
+      this.check(TokenKind.RealLiteral) ||
+      this.check(TokenKind.TrueLiteral) ||
+      this.check(TokenKind.FalseLiteral) ||
+      this.check(TokenKind.StringLiteral) ||
+      this.check(TokenKind.NilLiteral) ||
+      this.check(TokenKind.IntIdent) ||
+      this.check(TokenKind.RealIdent) ||
+      this.check(TokenKind.BoolIdent) ||
+      this.check(TokenKind.StringIdent) ||
+      this.check(TokenKind.LParen)
+    );
   }
 
-  private rmdir(): Instr {
-    const params = this.params();
-    return new Command('rmdir', params);
+  private checkExpressionStatementStart(): boolean {
```

Comment:
This is reasonable to start. But it's not current possible to call a command whose name happens to overlap with a keyword. Bash solves this by allowing the command name to be an expression. But here, we want bare strings to be interpreted as expression statements.

The most obvious way to cheese this would be to introduce a `cmd` keyword. But I'd like to think about this more before committing. Perhaps a draft ADR?